### PR TITLE
fix(dashboard): make Add Record dialog scroll with many columns

### LINKED ISSUE #1595: [Bug]: Add Record dialog overflows with many columns
## What's the bug?

The database Add Record dialog can overflow when a table has many editable columns. Users may be unable to reach lower fields or the footer actions because the dialog content is clipped instead of reliably scrolling.

## Validation

Valid from code inspection. `RecordFormDialog` renders one `RecordFormField` for every non-system column in `displayFields`, so tables with many columns create a tall form. The form body is wrapped in the dashboard `ScrollArea`, but the dialog itself uses `overflow-hidden` and the scroll area is only given `max-h-[500px]`; the shared `ScrollArea` viewport is `h-full`, which does not reliably create a scrollable viewport unless the root has a definite height. This matches the observed behavior where content overflows and no usable scrollbar appears.

Relevant code:

- `packages/dashboard/src/features/database/components/RecordFormDialog.tsx:41` filters system fields, leaving all other table columns as form fields.
- `packages/dashboard/src/features/database/components/RecordFormDialog.tsx:117` wraps the mapped fields in `<ScrollArea className="max-h-[500px]">`.
- `packages/dashboard/src/features/database/components/RecordFormDialog.tsx:119` maps every editable column into a `RecordFormField`.
- `packages/ui/src/components/Dialog.tsx:43` gives `DialogContent` `overflow-hidden`, so overflowing body content is clipped by the modal container.
- `packages/dashboard/src/components/radix/ScrollArea.tsx:12` sets the root to `overflow-hidden`, and `packages/dashboard/src/components/radix/ScrollArea.tsx:15` uses an `h-full` viewport, which needs a definite parent height to scroll consistently.

## How to reproduce

1. Open the dashboard database tables view.
2. Select or create a table with enough non-system columns to exceed the Add Record dialog height.
3. Click **Add Record**.
4. Observe that the fields overflow/clamp and there is no reliable vertical scrollbar for reaching the rest of the form.

## Expected Behavior

The Add Record dialog should constrain itself to the viewport and allow the field list to scroll, while keeping the header and footer actions reachable.

## Actual Behavior

With many columns, the dialog content can overflow/clamp inside the modal, and the user cannot reliably scroll to all fields or actions.

## Proposed Solution

Make the Add Record dialog a constrained flex column with a viewport-relative max height, and make the field list the scrolling region. For example:

- Add a max height such as `max-h-[calc(100dvh-2rem)]` to the dialog content/form.
- Give the form a `min-h-0` flex layout.
- Give the field-list scroll area a definite flex/height constraint such as `min-h-0 flex-1 overflow-y-auto`, or update the local `ScrollArea` usage so its viewport has a definite constrained height.

## Acceptance Criteria

- Add Record remains within the viewport on desktop and smaller screens.
- Tables with many editable columns show a usable vertical scrollbar in the dialog body.
- Header/close button and footer actions remain visible and usable.
- Add or update a component/UI test if practical for the dialog overflow behavior.

### PR DESCRIPTION
**Pull request by junaiddshaukat**
## What
Fixes #1595 — Add Record dialog overflowed the viewport with many columns; no usable scrollbar.

## Why
Centered `DialogContent` sizes to content (no max-height), and the field list's `ScrollArea` had only `max-h-[500px]`. The shared `ScrollArea` viewport is `h-full`, which needs a **definite** parent height to scroll — `max-h` alone is indefinite, so the body grew unbounded and clipped.

## Fix
Constrain the form to the viewport, make the field list the single scroll region (same pattern as `LinkRecordDialog`):
- form → `max-h-[calc(100dvh-2rem)] flex flex-col`
- header/footer → `shrink-0`
- field-list `ScrollArea` → `min-h-0 flex-1`

## Test
Added `RecordFormDialog.test.tsx` (component): many columns render, system fields filtered, footer Add Record stays reachable/submittable.
Run: `npm --workspace @insforge/dashboard run test:component -- RecordFormDialog`

> Note: layout/scroll isn't measurable in jsdom; the test guards the structural contract. `npx turbo run typecheck` has ~46 **pre-existing** errors on `main` (storage jest-dom matchers, socket.io types) unrelated to this change — none in touched 

---
## Summary by cubic
Fixes the Add Record dialog overflowing on wide tables by constraining form height and making only the field list scroll. Keeps the header and footer visible and actions clickable; addresses #1595.

- **Bug Fixes**
  - Form constrained to viewport: `max-h-[calc(100dvh-2rem)] flex flex-col`.
  - Header/footer set to not shrink: `shrink-0`.
  - Field list is the only scroll region: `ScrollArea` → `min-h-0 flex-1` (replaces `max-h-[500px]`).
  - Tests: added `RecordFormDialog.test.tsx` verifying many columns render, system fields are filtered, and "Add Record" stays submittable; updated to pass required `schemaName` prop in `@insforge/dashboard`.

<sup>Written for commit 25bceca77786ee65df65bead6efbd1137066fa62. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/InsForge/InsForge/pull/1600?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

> [!NOTE]
> ### Fix Add Record dialog to scroll when many columns are present
> - Updates layout classes in [`RecordFormDialog.tsx`](https://github.com/InsForge/InsForge/pull/1600/files#diff-d3187dbcafd16ec1a63148f51efa47c0be0ebf57ed4bb6b64842c3d434ed04e5): the `<form>` gets `max-h-[calc(100dvh-2rem)]` and flex-column layout, header/footer get `shrink-0`, and the `<ScrollArea>` switches to `min-h-0 flex-1` so it fills remaining space and scrolls.
> - Adds a new test file [`RecordFormDialog.test.tsx`](https://github.com/InsForge/InsForge/pull/1600/files#diff-599e8e63a394b3012fd768c3763148ff10167c9371ec6a5f97f1362e25b0694e) covering field filtering (system fields excluded) and form submission with many columns.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 25bceca.</sup>
> 
>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved the record form dialog layout so long forms remain usable: the dialog body now uses a full-height flex layout with a flex-based scroll area, keeping the footer and actions visible.
* **Tests**
  * Added React Testing Library + Vitest coverage to verify editable-field rendering (excluding system fields) and to confirm the “Add Record” flow calls record creation once and closes the dialog.

### CODE REVIEW

File: packages/dashboard/src/features/database/components/__tests__/RecordFormDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    columnName: `field_${index}`,
+    type: 'string',
+    isNullable: true,
+    isUnique: false,
+  }));
+  return [...systemColumns, ...editableColumns];
+}
+
+describe('RecordFormDialog', () => {
+  it('renders every editable column and filters system fields', () => {
```
Comment (Fermionic-Lyu) [inline packages/dashboard/src/features/database/components/__tests__/RecordFormDialog.test.tsx:30]: Both `render` calls omit the required `schemaName` prop (the interface declares `schemaName: string` and the component passes it to `useRecords(tableName, schemaName)`). The tests still run since esbuild strips types and `useRecords` is mocked, but `turbo run typecheck` will flag two new errors here — which contradicts the PR note that no touched files have typecheck errors. Pass `schemaName="public"` to both renders.

Comment (junaiddshaukat) [reply packages/dashboard/src/features/database/components/__tests__/RecordFormDialog.test.tsx:30]: Good catch — branch was based on pre-schemaName main. Synced with main and passed schemaName="public" to both renders. turbo run typecheck clean on touched files now.

**Review by Fermionic-Lyu — APPROVED** [review]
LGTM. Thanks for the PR!
